File: apps/web/src/types/market.ts
export type Direction="BULLISH"|"BEARISH"|"NEUTRAL";
export type Action="BUY AREA"|"SELL AREA"|"WAIT"|"NO TRADE";
export type Candle={time:number;open:number;high:number;low:number;close:number;volume:number};
export type Indicators={ema20:number;ema50:number;rsi:number;macd:number;signal:number;bbUpper:number;bbLower:number;atr:number;vwap:number};
export type Analysis={direction:Direction;action:Action;quality:number;entry:number;sl:number;tp:number;invalidation:number;setup:string;waiting:string;reason:string[];candle:string;structure:string;momentum:string};
export type AssetData={symbol:string;name:string;price:number;change:number;candles:Candle[];indicators:Indicators;analysis:Analysis};

File: services/api/app/schemas/market.py
from datetime import datetime
from pydantic import BaseModel, Field

class Candle(BaseModel):
    time: datetime
    open: float
    high: float
    low: float
    close: float
    volume: float = 0

class Quote(BaseModel):
    symbol: str
    price: float
    change: float | None = None
    change_percent: float | None = None
    timestamp: datetime

class MarketHistory(BaseModel):
    symbol: str
    interval: str
    candles: list[Candle]
    source: str

class IndicatorSnapshot(BaseModel):
    ema20: float | None = None
    ema50: float | None = None
    rsi: float | None = None
    macd: float | None = None
    macd_signal: float | None = None
    bb_upper: float | None = None
    bb_lower: float | None = None
    atr: float | None = None
    vwap: float | None = None

class MarketAnalysisRequest(BaseModel):
    symbol: str = Field(min_length=1, max_length=30)
    interval: str = "1d"
    period: str = "6mo"
    depth: str = "standard"

class PredictionPoint(BaseModel):
    time: datetime
    value: float
    lower: float | None = None
    upper: float | None = None

class AnalysisResponse(BaseModel):
    symbol: str
    current_price: float
    direction: str
    action: str
    confidence: float
    setup: str
    entry: float | None
    stop_loss: float | None
    take_profit: float | None
    invalidation: float | None
    indicators: IndicatorSnapshot
    prediction: list[PredictionPoint]
    sentiment: dict
    reasons: list[str]
    risk_notes: list[str]
